pilot/cmd/istioctl/gendeployment/cmd.go

```diff
@@ -0,0 +1,184 @@
+			"default control plane: Pilot, Mixer, CA, and Ingress Proxies, with mTLS enabled.",
+		Example: `istioctl gen-deploy --features routing,policy,initializer -o helm`,
+		RunE: func(c *cobra.Command, args []string) error {
+			if err := install.setFeatures(*features); err != nil {
+				return err
+			}
+			install.Namespace = *istioNamespaceFlag
+			switch strings.ToLower(out) {
+			case "helm":
+				_, err := fmt.Fprint(os.Stdout, valuesFromInstallation(install))
+				return err
+			case "yaml":
+				values, err := getValues(valuesPath, install)
+				if err != nil {
+					return err
+				}
+				rendered, err := yamlFromInstallation(values, *istioNamespaceFlag, helmChartLocation)
+				if err != nil {
+					return err
+				}
+				_, err = fmt.Fprint(os.Stdout, rendered)
+				return err
+			default:
+				return fmt.Errorf("unsupported output %q", out)
+			}
+		},
+	}
+
+	cmd.PersistentFlags().StringVar(&valuesPath, "values", "", "Path to the Helm values.yaml file used to render YAML "+
+		"deployments locally when --out=yaml. Flag values are ignored in favor of using the file directly.")
+
+	features = cmd.PersistentFlags().StringArrayP("features", "f", []string{},
+		`List of Istio features to enable. Accepts any combination of "mtls", "telemetry", "routing", "ingress", "policy", "initializer".`)
+	cmd.PersistentFlags().StringVar(&install.Hub, "hub", install.Hub, "The container registry to pull Istio images from")
+	cmd.PersistentFlags().StringVar(&install.MixerTag, "mixer-tag", install.MixerTag, "The tag to use to pull the `mixer` container")
+	cmd.PersistentFlags().StringVar(&install.PilotTag, "pilot-tag", install.PilotTag, "The tag to use to pull the `pilot-discovery` container")
+	cmd.PersistentFlags().StringVar(&install.CaTag, "ca-tag", install.CaTag, "The tag to use to pull the `ca` container")
+	cmd.PersistentFlags().StringVar(&install.ProxyTag, "proxy-tag", install.ProxyTag, "The tag to use to pull the `proxy` container")
+	cmd.PersistentFlags().BoolVar(&install.Debug, "debug", install.Debug, "If true, uses debug images instead of release images")
+	cmd.PersistentFlags().Uint16Var(&install.NodePort, "ingress-node-port", install.NodePort,
+		"If provided, Istio ingress proxies will run as a NodePort service mapped to the port provided by this flag. "+
+			"Note that this flag is ignored unless the \"ingress\" feature flag is provided too.")
+	cmd.PersistentFlags().StringVarP(&out, "out", "o", "helm", `Output format. Acceptable values are:
+					"helm": produces contents of values.yaml
+					"yaml": produces Kubernetes deployments`)
+
+	// TODO: figure out how we want to package up the charts with the binary to make this easy
+	cmd.PersistentFlags().StringVar(&helmChartLocation, "helm-chart-dir", ".",
+		"The directory to find the helm charts used to render Istio deployments. -o yaml uses these to render the helm chart locally.")
+
+	_ = cmd.PersistentFlags().MarkHidden("hub")
+	_ = cmd.PersistentFlags().MarkHidden("mixer-tag")
```

Comment:
Can we add an 'istio-tag' - and use it as a default value ? 

Can we add a more generic 'istio-release = daily | weekly  | stable ' - and auto-pull the tag ?

Reply:
So we need to support tags-per-component in quite a few spots where we do things with templates, but I'm on board with adding a higher level tag for users to interact with.

Can we punt to a follow on PR for that, as this PR is already getting very large?

Reply:
I'm also planning to tackle the feature naming stuff in a follow on. I'm in the process of reworking the helm templates too, and that all goes hand-in-hand.